modules/hal_nordic/CMakeLists.txt

```diff
@@ -11,7 +11,7 @@ if(CONFIG_NRF_REGTOOL_GENERATE_UICR)
   list(APPEND nrf_regtool_components GENERATE:UICR)
 endif()
 if(DEFINED nrf_regtool_components)
-  find_package(nrf-regtool 5.1.0
+  find_package(nrf-regtool 5.1.0 REQUIRED
```

Comment:
Commit message suggests that this should also be `EXACT`

Reply:
Does it make a difference? When checking this PR locally with 5.0.0 installed I get a fatal error:

```
CMake Error at /Users/carles/src/zephyr/zephyr/modules/hal_nordic/CMakeLists.txt:14 (find_package):
  Could not find a configuration file for package "nrf-regtool" that is
  compatible with requested version "5.1.0".

  The following configuration files were considered but not accepted:

    /Users/carles/src/zephyr/zephyr/modules/hal_nordic/nrf-regtool/nrf-regtoolConfig.cmake, version: 5.0.0
```

Reply:
`EXACT` would make the required version exactly 5.1.0. Without it, it's 5.1.0 or newer.

Reply:
Thanks. I will change the commit message, since we only want to require a minimum version.